src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/DiagnosticReportMapperTest.java

```diff
@@ -1,19 +1,234 @@
+                    () -> assertThat(coding.getDisplay()).isEqualTo("Diagnostic studies report"),
+                    () -> assertThat(coding.getSystem()).isEqualTo("http://snomed.info/sct")
+                )),
+            () -> assertThat(result.getStatus()).isEqualTo(DiagnosticReport.DiagnosticReportStatus.UNKNOWN),
+            () -> assertThat(result.getIssued()).isEqualTo(Date.from(dateIssued.toInstant()))
+        );
+    }
+
+    @Test
+    void testMapToDiagnosticReportWithSubjectResource() {
+        final Message message = new Message(List.of(
+            "S02+02",
+            "GIS+N",
+            "RFF+SRI:15/CH000042P/200010191704",
+            "STS++UN",
+            "DTM+ISR:201002251541:203"
+        ));
+
+        final var patient = mock(Patient.class);
+        when(resourceFullUrlGenerator.generate(any(Patient.class))).thenReturn("patient-full-url");
+
+        final var result = mapper.mapToDiagnosticReport(message, patient, Collections.emptyList(),
+            Collections.emptyList(), null, null, null);
+
+        assertThat(result.getSubject())
+            .extracting(Reference::getReference)
+            .isEqualTo("patient-full-url");
+    }
+
+    @Test
+    void testMapToDiagnosticReportWithSpecimenResources() {
+        final Message message = new Message(List.of(
+            "S02+02",
+            "GIS+N",
+            "RFF+SRI:15/CH000042P/200010191704",
+            "STS++UN",
+            "DTM+ISR:201002251541:203"
+        ));
+
+        final var specimen1 = mock(Specimen.class);
+        final var specimen2 = mock(Specimen.class);
+
+        when(resourceFullUrlGenerator.generate(nullable(Specimen.class))).thenReturn("some-specimen-id");
+        final var result = mapper.mapToDiagnosticReport(message, null, List.of(specimen1, specimen2),
+            Collections.emptyList(), null, null, null);
+
+        assertThat(result.getSpecimen())
+            .hasSize(2)
+            .extracting(Reference::getReference)
+            .containsExactly("some-specimen-id", "some-specimen-id");
+    }
+
+    @Test
+    void testMapToDiagnosticReportWithObservationResources() {
+        final Message message = new Message(List.of(
+            "S02+02",
+            "GIS+N",
+            "RFF+SRI:15/CH000042P/200010191704",
+            "STS++UN",
+            "DTM+ISR:201002251541:203"
+        ));
+
+        final var observation1 = mock(Observation.class);
+        final var observation2 = mock(Observation.class);
+
+        when(resourceFullUrlGenerator.generate(nullable(Observation.class))).thenReturn("some-observation-id");
+        final var result = mapper.mapToDiagnosticReport(message, null, Collections.emptyList(),
+            List.of(observation1, observation2), null, null, null);
+
+        assertThat(result.getResult())
+            .hasSize(2)
+            .extracting(Reference::getReference)
+            .containsExactly("some-observation-id", "some-observation-id");
```

Comment:
Should we alter this test to use a different ID for each observation, to show that the mapper doesn't just duplicate the ID?

Reply:
I'm not sure how to do this with a _nullable_ type as the mocks `observation1` and `observation2` can be nullable which means that I get a `strict stubbing mismatch` error when I try and do something like:
```
when(resourceFullUrlGenerator.generate(specimen1)).thenReturn("some-observation-id-1");
```

Reply:
@saileshp-kainos one of the options would be to use lenient() and have something like this:
```
lenient().when(resourceFullUrlGenerator.generate(observation1)).thenReturn("some-observation-id1");
lenient().when(resourceFullUrlGenerator.generate(observation2)).thenReturn("some-observation-id2");
```

However, I am not sure if it's a good approach or not

Reply:
I figured out what was causing it. 

I needed to add another `when` so it stubs the value for `resourceFullUrlGenerator.generate(nullable(Patient.class))` so I can continue with the rest of the class to stub the appropriate values of `observation1` and `observation2`. 

Thanks @antonlebedjko & @graham-andrew for pointing this out 🥳  I'll commit this once I've done it for the `Observation` and `Specimen` tests

Reply:
Done in c81e258